cloud-db/acl-overview/android.md

```diff
@@ -0,0 +1,242 @@
+
+
+These are the concepts you need to know:
+- Record ownership, **the owner has full access** to the record.
+- The 2 types of ACL in Skygear: [Record-based ACL][doc-record-acl] and [Field-based ACL][doc-field-acl].
+
+:::tips
+When you should use Record-based ACL or Field-based ACL?
+
+If every single record has the same ACL, that the same  have the same access rights to a record, you can use Record-based ACL.
+
+Otherwise, if a set of users have has different access rights for each field of a record, you would need to use Field-based ACL.
+:::
+
+Before diving into Record-based ACL and Field-based ACL, let's learn about the basic concepts of Skygear ACL.
+
+## ACL Access Level
+
+There are 3 access levels in Skygear ACL:
+
+- No Access
+- Read Write Access
+- Read Only Access
+
+This is how you can set it:
+
+```java
+AccessControl acl = new AccessControl();
+
+acl.addEntry(new AccessControl.Entry(AccessControl.Level.NO_ACCESS));
+acl.addEntry(new AccessControl.Entry(AccessControl.Level.READ_ONLY));
+acl.addEntry(new AccessControl.Entry(AccessControl.Level.READ_WRITE));
+
+// and more...
+```
+
+:::tips
+
+**Read access**
+
+Read access grants users right to *query* and *fetch* records, which includes getting all the fields of a record as well as the ACL of the record.
+
+
+**Write access**
+
+Write access grants users right to *save* and *delete* records, which includes adding,
+updating and removing all the fields (**EXCEPT** [reserved columns][doc-reserved-columns]) of a record as well as the ACL of the record.
+
+:::
+
+## ACL User Target
+
+User Target defines *who* can perform the specified actions. It can be a specific user or a group of users. Below are the types of User Target you can define.
+
+- **Public**: refers to all other users of the app, unauthenticated (not logged in) users included
+- **By User**: users specified by User ID
+- **By Role**: users grouped into specific roles by admin
+
+Below we will talk about how to create roles and assign a role to users.
+
+### Role Configuration
+
+#### Define role
+
+You can create different roles and later use them to design ACL targeting users of specific roles.
+
+```java
+Role manager = new Role("manager");
+Role visitor = new Role("visitor");
+```
+
+#### Admin role
+
+In Skygear roles are either admin role or non-admin role. By default, there is an admin role called `Admin`.
+
+What only admin role can do:
+
+- Assign role to user
+- Revoke role to user
+
+:::tips
+For security concern, only admin can change the role of a user, thus the user must request another user with admin role to change the role for him.
+
+Do not abuse role to model user grouping. If a group can be joined by other users without requesting permission, that should be a public group, which does not require ACL.
+:::
+
+
+#### Setting admin role and default role in debug mode
+
+You define new roles to be admin role.
```

Comment:
I think it better to mention the setting should be done during the development, not production.

Reply:
Agree, I will use `:::caution` for this.